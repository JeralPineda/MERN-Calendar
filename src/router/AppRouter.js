import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';

import { startChecking } from '../actions/auth';
import { LoginScreen } from '../components/auth/LoginScreen';
import { CalendarScreen } from '../components/calendar/CalendarScreen';
import { Loader } from '../components/loader/Loader';

export const AppRouter = () => {
   const { checking } = useSelector((state) => state.auth);

   const dispatch = useDispatch();

   useEffect(() => {
      dispatch(startChecking());
   }, [dispatch]);

   if (checking) {
      return <Loader />;
   }

   return (
      <Router>
         <div>
            <Switch>
               <Route exact path='/login' component={LoginScreen} />

               <Route exact path='/' component={CalendarScreen} />

               <Redirect to='/' />
            </Switch>
         </div>
      </Router>
   );
};
