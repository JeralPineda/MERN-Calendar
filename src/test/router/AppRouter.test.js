import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { AppRouter } from '../../router/AppRouter';

// jest.mock('../../../actions/events', () => ({
//    eventStartDelete: jest.fn(),
// }));

const middleware = [thunk];
const mockStore = configureStore(middleware);

// store.dispatch = jest.fn();

describe('Pruebas en <AppRouter />', () => {
   test('debe de mostrar el loader', () => {
      // para que la prueba pase, por el #root no dejaba pasar la prueba
      const initState = {
         auth: {
            checking: true,
         },
      };

      let store = mockStore(initState);

      const wrapper = mount(
         <Provider store={store}>
            <AppRouter />
         </Provider>
      );

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('.loader').exists()).toBe(true);
   });
});
