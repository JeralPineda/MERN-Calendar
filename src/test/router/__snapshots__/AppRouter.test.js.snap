// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar el loader 1`] = `
<div
  className="loader-background"
>
  <div
    className="loader"
  />
</div>
`;
